replica/table.cc

```diff
@@ -634,11 +635,25 @@ table::update_cache(compaction_group& cg, lw_shared_ptr<memtable> m, std::vector
         m->mark_flushed(std::move(new_ssts_ms));
         try_trigger_compaction(cg);
     });
+    std::exception_ptr ex;
     if (cache_enabled()) {
+      try {
         co_return co_await _cache.update(std::move(adder), *m);
-    } else {
-        co_return co_await _cache.invalidate(std::move(adder)).then([m] { return m->clear_gently(); });
+      } catch (...) {
+        ex = std::current_exception();
+      }
+    }
+    co_await _cache.invalidate(std::move(adder));
```

Comment:
You only want to invalidate the cache when an update fails or if cache is disabled.
But here you are invalidating it unconditionally, even on the normal path.

That's quite a catastrophic mistake, yet CI didn't catch it. I wonder if we have any test in the release pipeline able to catch it.

Reply:
The normal path, with `cache_enabled`, does `co_return co_await _cache.update`,
meaning it won't get to `co_await _cache.invalidate`.

The latter is executed if `!cache_enabled()` or if `_cache.update` throws.

Reply:
Ah, right. Sorry. I didn't register the `co_return`.